chefkoch/recipe.py

```diff
@@ -0,0 +1,771 @@
+        except KeyError as err:
+            raise KeyError("The start, stop or step field of " + str(entry) +
+                            "are missing.")
+        valid_start = isinstance(i, int) or isinstance(i, float)
+        valid_stop = isinstance(stop, int) or isinstance(stop, float)
+        valid_step = isinstance(step, int) or isinstance(step, float)
+        valid = valid_start and valid_step and valid_stop
+        if not valid:
+            logger.warning("The start, step and stop value of a parameter range" +
+                           " need to by of type int or float, so an empty list" + 
+                           " was appended! Check for correctness!")
+            return
+        # get the direction of the range before using <= or >=
+        if i < stop and step > 0:
+            # add all values within range
+            while i <= stop:
+                logger.debug("Adding value " + str(i))
+                self.values.append(i)
+                i = i + step
+        elif i > stop and step < 0:
+            # add all values within range
+            while i >= stop:
+                logger.debug("Adding value " + str(i))
+                self.values.append(i)
+                i = i + step
+        else:
+            logger.warning("The start " + str(i) + ", stop " + str(stop) + " and step " +
+                           str(step) + " of the range leave an empty list for" +
+                          " this paramter. Please check if this is intended.")
+
+    def appendEntry(self, entry):
+        """
+        Appends a single entry within the JSON data received from the flavour file.
+        
+        :param entry:       file or range or any other value
+        :type entry:        dictionary
+        """
+        try:
+            type = entry['type']
+        except KeyError as err:
+            logger.debug("There is no type specified to this entry. It will" +
+                         " be interpreted and appended as a dictionary.")
+            self.values.append(entry)
+            return
+        except TypeError as err:
+            # entry is not a dictionary
+            logger.debug("The entry is not a dictionary. It is appended normally.")
+            self.values.append(entry)
+            return
+        # if there is a type to the entry, test if it is known
+        logger.debug("It is of type " + type)
+        if type == 'file':
+            try:
+                self.appendFileParam(entry)
+            # todo: capture and handle this exception inside appenFileParam
+            except ValueError as err:
+                pass
+        elif type == 'range':
+            self.appendValuesFromRange(entry)
+        else:
+            # allow everything else by default, value could also be a list
+            logger.warning("There is a type specified to " + str(entry) +
+                           ", but neither 'range' nor 'file'. It will be " +
+                           "appended as a dictionary.")
```

Comment:
Use string formatting for such purposes
`"string %s." %(str(entry), )`

Note, that what comes after the `%` operator must be a tuple. So if there's only one element to it, do not forget the trailing comma or the tuple disappears and the one element will actually be iterated by format!